scipy/signal/_signaltools.py

```diff
@@ -2167,16 +2167,20 @@ def deconvolve(signal, divisor):
     array([0, 2, 1, 0, 2, 3, 1, 0, 0])
     >>> recovered, remainder = signal.deconvolve(recorded, impulse_response)
     >>> recovered
-    array([ 0.,  1.,  0.,  0.,  1.,  1.,  0.,  0.])
+    array([ 0., 1., 0., 0., 1., 1., 0., 0.])
 
     See Also
     --------
     numpy.polydiv : performs polynomial division (same operation, but
-                    also accepts poly1d objects)
+    also accepts poly1d objects)
 
     """
     num = np.atleast_1d(signal)
     den = np.atleast_1d(divisor)
+    if num.ndim > 1:
+        raise ValueError("signal must be 1-D.")
+    if den.ndim > 1:
+        raise ValueError("divisor must be 1-D.")
```

Comment:
I am not sure if we need 2 checks. I would have combined both. Let see what others say.

Reply:
This increases the specificity of the error, which is nice for the user. I'd keep them separate

Reply:
Thanks for the advices I intended the same as @larsoner says